src/commands/Minion/alch.ts

```diff
@@ -0,0 +1,137 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { BotCommand } from '../../lib/BotCommand';
+import getOSItem from '../../lib/util/getOSItem';
+import { Util } from 'oldschooljs';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import resolveItems from '../../lib/util/resolveItems';
+import hasItemEquipped from '../../lib/gear/functions/hasItemEquipped';
+import { bankHasAllItemsFromBank, removeBankFromBank, resolveNameBank, roll } from '../../lib/util';
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+
+const options = {
+	max: 1,
+	time: 10_000,
+	errors: ['time']
+};
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 1,
+			usage: '[quantity:int{1}] <itemName:...string>',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, itemName]: [number | undefined, string]) {
+		const { id: itemId, highalch, name } = getOSItem(itemName);
+		const numOfItems = await msg.author.numberOfItemInBank(itemId);
+
+		if (numOfItems === 0) throw `'You don't have any of this item to alch'`;
+
+		if (!quantity) {
+			quantity = numOfItems;
+		}
+
+		let fireRuneCost = quantity * 5;
+		let natureRuneCost = quantity;
+		let savedRunes = 0;
+
+		const unlimitedFireRuneProviders = resolveItems([
+			'Staff of fire',
+			'Fire battlestaff',
+			'Mystic fire staff',
+			'Lava battlestaff',
+			'Mystic lava staff',
+			'Steam battlestaff',
+			'Mystic steam staff',
+			'Smoke battlestaff',
+			'Mystic smoke staff',
+			'Tome of fire (empty)',
+			'Tome of fire'
+		]);
+
+		for (const runeProvider of unlimitedFireRuneProviders) {
+			if (
+				hasItemEquipped(runeProvider, msg.author.settings.get(UserSettings.Gear.Skilling))
+			) {
+				fireRuneCost = 0;
+				break;
+			}
+		}
+
+		const { id: bryophytasStaffId } = getOSItem("Bryophyta's staff");
+
+		if (
+			hasItemEquipped(bryophytasStaffId, msg.author.settings.get(UserSettings.Gear.Skilling))
+		) {
+			for (let i = 0; i < quantity; i++) {
+				if (roll(15)) {
+					natureRuneCost--;
+					savedRunes++;
+				}
+			}
+		}
+
+		const alchValue = quantity * highalch;
+
+		const consumedItems = resolveNameBank({
+			...(fireRuneCost > 0 ? { 'Fire rune': fireRuneCost } : {}),
+			'Nature rune': natureRuneCost,
+			[itemName]: quantity
```

Comment:
```suggestion
			[name]: quantity
```

getOSItem cleans the name passed to it so it could technically not be the true name of the item so use the returned OSItems name